libs/models/datapoints.py

```diff
@@ -0,0 +1,400 @@
+import datetime
+from abc import ABC, abstractmethod
+from logging import Logger
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, root_validator, validator
+
+from electricitymap.contrib.config import EXCHANGES_CONFIG, ZONES_CONFIG, ZoneKey
+from electricitymap.contrib.libs.models.constants import VALID_CURRENCIES
+
+LOWER_DATETIME_BOUND = datetime.datetime(2000, 1, 1, tzinfo=datetime.timezone.utc)
+
+
+class Mix(BaseModel, ABC):
+    @root_validator
+    def _validate_mix(cls, values: Dict[str, Optional[float]]):
+        if all(v is None for v in values.values()):
+            raise ValueError("Mix is completely empty")
+
+
+class ProductionMix(Mix):
+    biomass: Optional[float]
+    coal: Optional[float]
+    gas: Optional[float]
+    geothermal: Optional[float]
+    hydro: Optional[float]
+    nuclear: Optional[float]
+    oil: Optional[float]
+    solar: Optional[float]
+    unknown: Optional[float]
+    wind: Optional[float]
+
+
+class StorageMix(Mix):
+    battery: Optional[float]
+    hydro: Optional[float]
+
+
+class Datapoint(BaseModel, ABC):
+    zoneKey: ZoneKey
+    datetime: datetime
+    source: dict
```

Comment:
If source is the datasource should this not be a string or do we plan on changing the data structure to be able to return multiple sources?

Reply:
do we want to include the source as dict in the first instance of this project?

Reply:
I'm wondering if this might not fall outside of the scope for the first version of this project, where we want to make sure that data points loaded to `parser_table` are valid.

Reply:
I reverted it back to str for this PR, I put the improved source capabilities on another branch for a subsequent PR.